### PR TITLE
Improve redundancy reporting

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
This PR improves the redundancy report of the renter endpoint.
The renter doesn't count contracts that are not good for renewal towards redundancy while the renter endpoint does. That's why the renter might report redundancies > 3x.
Unfortunately we can't simply change the renter endpoint's behavior to match the renter's internal behavior since that might lead to redundancies < 1x which might lead to a poor UX even though the files can still be downloaded.
That's why this PR changes the endpoint to calculate both redundancies and if the redundancy without `goodForRenew` contracts drops below 1 it reports the higher redundancy.

### CODE REVIEW

File: modules/renter/files.go [hunk 1]
```diff
@@ -239,6 +264,11 @@ func (r *Renter) FileList() []modules.FileInfo {
 		offline := r.hostContractor.IsOffline(id)
 		return offline
 	}
+	goodForRenew := func(id types.FileContractID) bool {
+		id = r.hostContractor.ResolveID(id)
+		cu, ok := r.hostContractor.ContractUtility(id)
```
Comment (DavidVorick) [inline modules/renter/files.go]: should we expand the `cu` object to also include the `IsOffline` information? That will limit the number of times that we are calling out of package

**Comment by DavidVorick** [comment]
Overall looks good to me. My guess is a lot of this will change once we update the file structure, we should no longer need to loop through every chunk every time we want the redundancy of a file.

**Comment by ChrisSchinnerl** [comment]
@DavidVorick I merged the `isOffline` and `goodForRenew` methods into a `contractStatus` method that returns both.

File: modules/renter/files.go [hunk 2]
```diff
@@ -147,21 +148,46 @@ func (f *file) redundancy(isOffline func(types.FileContractID) bool) float64 {
 		return -1
 	}
 	for _, fc := range f.contracts {
+		offline, gfr := contractStatus(fc.ID)
```
Comment (DavidVorick) [inline modules/renter/files.go]: gfr is not readable to someone unfamiliar with the code

File: modules/renter/files.go [hunk 3]
```diff
@@ -88,10 +88,10 @@ func (f *file) numChunks() uint64 {
 }
 
 // available indicates whether the file is ready to be downloaded.
-func (f *file) available(isOffline func(types.FileContractID) bool) bool {
+func (f *file) available(contractStatus func(types.FileContractID) (bool, bool)) bool {
```
Comment (DavidVorick) [inline modules/renter/files.go]: the return values here should probably be named, confusing to have a `(bool, bool)) bool` construction

File: modules/renter/files.go (hunk 2 above)
Comment (ChrisSchinnerl) [reply modules/renter/files.go]: done

File: modules/renter/files.go (hunk 3 above)
Comment (ChrisSchinnerl) [reply modules/renter/files.go]: done

**Pull request merged**